### PR TITLE
Update documentation

### LINKED ISSUE #227: [DOC]: Add more content to the API references page and update list of planned features
### Describe your documentation update
* Update API References page with more information on how to use the APIs
* Clean up the planned features list 
* Update links to external resources such as `xarray`, `dask`, `cf_xarray`, and cf conventions

Related to #224

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

- Closes #227 

Summary of Changes
- Update API Reference page with an overview section and more content covering the accessor classes
- Update FAQs page with section on data wrangling and how to handle data quality issues
- Update planned features section of homepage and README
- Update docstrings of public APIs
- Make some `dataset.py` functions private and remove them from API Reference page
- Add `sphinx-autobuild` to `dev.yml` to make documentation update and build cycle quicker
- Removed link to Xarray API References on the side bar -- the user should be able to find this themselves if they are using `xarray`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: docs/faqs.rst [hunk 1]
```diff
@@ -2,21 +2,28 @@
 Frequently Asked Questions
 ==========================
 
-How do I open files that have conflicting values for a data variable(s)?
-------------------------------------------------------------------------
-In xarray, the default setting for checking compatibility across files is ``compat='no_conflicts'``.
+Data Wrangling
+--------------
+
+Some datasets might have quality issues, such as non-CF compliant attributes and inconsistent values.
```
Comment (tomvothecoder) [inline docs/faqs.rst:8]:
```suggestion
Some datasets might have data quality issues such as inconsistent floating point values between files or non-CF compliant attributes.
```

File: docs/faqs.rst [hunk 2]
```diff
@@ -2,21 +2,28 @@
...
+As a result, ``xarray`` and ``xcdat`` not being able to open these datasets using the default settings.
```
Comment (tomvothecoder) [inline docs/faqs.rst:9]:
```suggestion
If data quality issues are present, `xarray`` and ``xcdat`` might not be able to open the datasets.

You might be able to workaround data quality issues by:
1. Configuring ``open_dataset()`` or ``open_mfdataset()`` keyword arguments based on your needs.
2. Writing a custom `preprocess()` function to feed into ``open_mfdataset()``, which preprocesses each dataset file individually before joining them into a single Dataset object.

``xcdat`` aims to be generalized library, so we don't intend on implementing functionality to fix poor data for limited use cases.
```

**Review by tomvothecoder — COMMENTED** [review]
Some review fixes

**Comment by tomvothecoder** [comment]
Hi @jasonb5 and @lee1043, I updated the `xcdat` documentation so it is hopefully clearer.

Please check out the following pages and let me know if you have feedback. Thanks!

Homepage
- Old version: https://xcdat.readthedocs.io/en/latest/#
- New version: https://xcdat.readthedocs.io/en/docs-227-guides/

Getting Started:
- Old version: https://xcdat.readthedocs.io/en/latest/getting_started.html
- New version: https://xcdat.readthedocs.io/en/docs-227-guides/getting_started.html

API Reference
- Old version: https://xcdat.readthedocs.io/en/latest/api.html
- New version https://xcdat.readthedocs.io/en/docs-227-guides/api.html

FAQs
- Old version: https://xcdat.readthedocs.io/en/latest/faqs.html
- New Version: https://xcdat.readthedocs.io/en/docs-227-guides/faqs.html

**Comment by tomvothecoder** [comment]
Just an FYI @pochedls and @chengzhuzhang if you are interested. @pochedls already does enough review so I didn't tag him this time!

**Comment by lee1043** [comment]
@tomvothecoder the update document is looking great, thank you for your effort. 

One minor suggestion: I see you've removed [From source](https://xcdat.readthedocs.io/en/latest/getting_started.html#from-sources) installation info. I agree on such installation is not recommended by us, and perhaps not being used by users because conda is much easier. But can we still have that info somewhere in docs? No need to be highlighted, but could be hidden somewhere, so if needed one can dig out that info.

**Review by lee1043 — APPROVED** [review]

**Comment by tomvothecoder** [comment]
Thanks for reviewing @lee1043.

> One minor suggestion: I see you've removed [From source](https://xcdat.readthedocs.io/en/latest/getting_started.html#from-sources) installation info. I agree on such installation is not recommended by us, and perhaps not being used by users because conda is much easier. But can we still have that info somewhere in docs? No need to be highlighted, but could be hidden somewhere, so if needed one can dig out that info.

I decided to remove instructions on installing from source because I noticed other packages like `xarray` and `cf_xarray ` don't list it. Also, there will be some new dependencies for `xcdat` in #164 that are unavailable through  PyPi/pip (`esmpy`), which means running `python setup.py` results in missing dependencies.

**Comment by lee1043** [comment]
> I decided to remove instructions on installing from source because I noticed other packages like `xarray` and `cf_xarray ` don't list it. Also, there will be some new dependencies for `xcdat` in #164 that are unavailable through PyPi/pip (`esmpy`), which means running `python setup.py` results in missing dependencies (unless users install the missing dependencies themselves in the conda env).

@tomvothecoder thanks for checking. This makes good sense to me. I think that is a good reason to not include `from source`.

File: README.rst [hunk 3]
```diff
@@ -49,45 +49,52 @@ It serves as a spiritual successor to the Community Data Analysis Tools (`CDAT`_
...
-  - Calculate time series averages, climatologies, and departures
+  - Time series averages, climatologies, and departures
   - Weighted or unweighted
   - Optional centering of time coordinates using time bounds
+  - Optional seasonal configuration
 
-- Geospatial weighted averaging over rectilinear grid
+- Geospatial weighted averaging
 
+  - Supports rectilinear grid
```
Comment (tomvothecoder) [inline README.rst:76]:
```suggestion
  - Support rectilinear grid
```

File: README.rst [hunk 4]
```diff
@@ -49,45 +49,52 @@ It serves as a spiritual successor to the Community Data Analysis Tools (`CDAT`_
...
   - Optional specification of regional domain
 
 - Horizontal and vertical structured regridding
 
-  - Operate on rectilinear and cuvilinear grids
+  - Supports rectilinear and cuvilinear grids
```
Comment (tomvothecoder) [inline README.rst:81]:
```suggestion
  - Support rectilinear and cuvilinear grids
```

File: docs/faqs.rst [hunk 5]
```diff
@@ -2,21 +2,33 @@
...
+Examples of data quality issues include conflicting floating point values between files or non-CF compliant attributes.
 
-1. ``compat="override"``: skip comparing and pick variable from first dataset
-2. ``join="override"``:  if indexes are of same size, rewrite indexes to be those of the first object with that dimension. Indexes for the same dimension must have the same size in all objects.
+A few workarounds include:
+
+1. Configuring ``open_dataset()`` or ``open_mfdataset()`` keyword arguments based on your needs.
+2. Writing a custom ``preprocess()`` function to feed into ``open_mfdataset()``. This function preprocesses each dataset file individually before joining them into a single Dataset object.
+
+
```
Comment (tomvothecoder) [inline docs/faqs.rst:18]:
```suggestion
```

File: docs/index.rst [hunk 6]
```diff
@@ -10,45 +10,52 @@ It serves as a spiritual successor to the Community Data Analysis Tools (`CDAT`_
...
-- Geospatial weighted averaging over rectilinear grid
+- Geospatial weighted averaging
 
+  - Supports rectilinear grid
   - Optional specification of regional domain
 
 - Horizontal and vertical structured regridding
 
-  - Operate on rectilinear and cuvilinear grids
+  - Supports rectilinear and cuvilinear grids
```
Comment (tomvothecoder) [inline docs/index.rst:42]:
```suggestion
  - Support rectilinear and cuvilinear grids
```

**Pull request merged**